client/src/lib.rs

```diff
@@ -153,31 +150,76 @@ impl TurnkeyClient {
         }
     }
 
-    pub fn current_timestamp(&self) -> String {
+    /// Returns the current timestamp as a u128.
+    pub fn current_timestamp(&self) -> u128 {
         let now = SystemTime::now();
 
         now.duration_since(UNIX_EPOCH)
             .expect("Time went backwards")
             .as_millis()
-            .to_string()
+    }
+
+    /// Processes a `Request` (at `path`) by:
+    /// * Serializing the request to JSON
+    /// * Signing the request with the client's API key
+    /// * POSTing the POST body and the associated stamp to the Turnkey API
+    ///
+    /// This function is generic and can handle POSTing queries or activities.
+    pub async fn process_request<Request, Response>(
+        &self,
+        request: &Request,
+        path: String,
+    ) -> Result<Response, TurnkeyClientError>
+    where
+        Request: Serialize,
+        Response: DeserializeOwned,
+    {
+        let url = format!("{}{}", self.base_url, path);
+        let post_body = serde_json::to_string(&request)?;
+        let stamp = self.api_key.stamp(post_body.clone())?;
+        let res = self
+            .http
+            .post(url)
+            .header("X-Stamp", stamp)
+            .body(post_body)
+            .send()
+            .await?;
+
+        let status = res.status();
+        let text = res.text().await?;
```

Comment:
Is there a max size or timeout on the response size?

Reply:
Oh yeah, great point. I'm not specifying anything. So if there is a max size or timeout, it'll be the `reqwest` defaults.

Reply:
This led me down a rabbit-hole: which settings should we support "first-party" and which settings should we not? In order to allow people to override arbitrary things in the reqwest client I've opted for a builder pattern which gives access to the reqwest:ClientBuilder object;

One thing that's really neat is that enabling debug logging for the unerlying reqwest client becomes possible out of the box! Dropped a note about this in the README (which I moved from the top-level crate)